### PR TITLE
Remove support for `fields` in the bulk annotation API

### PR DESCRIPTION
**Pull request by marcospri**
Removing support for specifying fields on the annotation bulk API.

This is a nice feature that the service provides, that the API hints at supporting but that we have code to disable/limit. 

We could support this in the future if needed but the current approach, IMO, adds additional complexity without any of the benefits of the feature.

In addition this was making adding new fields a bit more complex than necessary.

## Testing

Same as https://github.com/hypothesis/h/pull/8166

### CODE REVIEW

File: h/views/api/bulk/annotation_schema.json [hunk 1]
```diff
@@ -15,31 +15,20 @@
...
+    "additionalProperties": true,
 
     "$defs": {
-        "fields": {
-            "title": "Fields",
-            "description": "A list of fields to include in the output",
-
-            "type": "array",
-            "items": {"type": "string"},
-            "enum": [["author.username", "group.authority_provided_id"]]
```
Comment (marcospri) [inline h/views/api/bulk/annotation_schema.json:37]: Note the enum here it only has one value, the two column in that order.

File: h/services/bulk_annotation.py [hunk 2]
```diff
@@ -58,19 +55,19 @@ def date_match(column: sa.Column, spec: dict):
     return sa.and_(*clauses)
 
 
+@dataclass
+class BulkAnnotation:
```
[the remark this answers is not in the document]
Comment (seanh) [reply h/services/bulk_annotation.py:58]: :+1:

File: h/services/bulk_annotation.py [hunk 3]
```diff
@@ -58,19 +54,19 @@ def date_match(column: sa.Column, spec: dict):
...
 
     # Aliases to distinguish between different types of user
     _AUTHOR = sa.orm.aliased(User, name="author")
     _AUDIENCE = sa.orm.aliased(User, name="audience")
 
-    # Acceptable values to pass in as `fields`
-    _FIELDS = {
-        "author.username": _AUTHOR.username,
-        "group.authority_provided_id": Group.authority_provided_id,
-    }
```
Comment (seanh) [inline h/services/bulk_annotation.py:72]: So the only acceptable fields were `"author.username"` and `"group.authority_provided_id"`

File: h/services/bulk_annotation.py [hunk 4]
```diff
@@ -82,51 +78,44 @@ def annotation_search(
...
-        if fields is None:
-            query = sa.select(Annotation)
-        else:
-            query = sa.select(list(cls._field_names_to_columns(fields))).select_from(
-                Annotation
-            )
+        columns = [
+            cls._AUTHOR.username,
+            Group.authority_provided_id,
+        ]
```
Comment (seanh) [inline h/services/bulk_annotation.py:116]: So it's now hard-coded to return `username` and `authority_provided_id` fields, which were previously the only two fields that were acceptable

File: h/services/bulk_annotation.py [hunk 5]
```diff
@@ -82,51 +78,44 @@ def annotation_search(
...
+
+        query = sa.select(columns).select_from(Annotation)
```
Comment (seanh) [inline h/services/bulk_annotation.py:118]: Could maybe just inline `columns`, it's only used in one place

```suggestion
        query = sa.select([cls._AUTHOR.username, Group.authority_provided_id]).select_from(Annotation)
```

File: h/views/api/bulk/annotation.py [hunk 6]
```diff
@@ -33,41 +33,28 @@ def bulk_annotation(request):
...
         raise ValidationError(str(err)) from err
 
-    present_row = _get_present_row(fields)
-    return get_ndjson_response((present_row(row) for row in annotation_rows))
-
-
-def _get_present_row(fields):
-    # The schema enforces this, but we currently rely on it being hard coded
-    # so some belt and braces is in order. For example, the exact order matters
-    assert fields == ["author.username", "group.authority_provided_id"]
```
[the remark this answers is not in the document]
Comment (seanh) [reply h/views/api/bulk/annotation.py:63]: Ugh, validation like this should have been happening in a validation schema not buried in a private helper method

**Review by seanh — APPROVED** [review]

**Pull request merged**